### ISSUE TITLE
SAML authentication in revel

### ISSUE THREAD

**Issue by GururajaUN** [revel/revel#1485]
I am using revel as framework for the application development. I need to enable user authentication with Okta IDP ( using SAML ).. 

I tried integrate "https://github.com/crewjam/saml" but it seems to be a big effort !

Any suggestions on the package / module that can be used for authentication in revel please ?

**Comment by ptman** [comment]
I used https://github.com/russellhaering/gosaml2/ with success

**Comment by GururajaUN** [comment]
Thanks ptman. I integrated https://github.com/crewjam/saml with success. Implemented the library as a proxy in front of revel so that only authenticated users are serviced by revel api's.

**Comment by ptman** [comment]
Ok, but that forces SAML on everyone, while we are using SAML as one possible auth method.

**Comment by GururajaUN** [comment]
Yes, wanted to enforce SAML on everyone and it was supposed to be only auth method.
Curious to understand how you integrate multiple auth modules. Any pointers for it please ?
